### PR TITLE
Create a library of gadgets

### PR DESCRIPTION
**Pull request by Acentelles**

### CODE REVIEW

File: taiga_halo2/examples/sudoku/app/sudoku_app_vp.rs [hunk 1]
```diff
@@ -874,133 +877,6 @@ impl StateUpdateConfig {
...
-        second.copy_advice(|| "second", region, self.advice[1], offset)?;
-        third.copy_advice(|| "third", region, self.advice[2], offset)?;
-        let value = first.value() * second.value() * third.value();
-
-        region.assign_advice(|| "out", self.advice[0], 1, || value)
-    }
-}
-
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub struct ValueCheckConfig {
```
Comment (XuyangSong) [inline taiga_halo2/examples/sudoku/app/sudoku_app_vp.rs:932]: `ValueCheckConfig` might not go to gadget lib. It's specific for the sudoku circuit.

**Review by XuyangSong — APPROVED** [review]

File: taiga_halo2/examples/taiga_sudoku/main.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+pub mod app_vp;
+pub mod dealer_intent_vp;
+pub mod gadgets;
+pub mod intent_app_vp;
+use crate::app_vp::halo2_sudoku_app_vp_circuit_final;
+fn main() {
+    halo2_sudoku_app_vp_circuit_final();
```
Comment (XuyangSong) [inline taiga_halo2/examples/taiga_sudoku/main.rs:7]: We may need a sudoku final transaction here later.
Let's merge it first.

**Review by XuyangSong — APPROVED** [review]
lgtm

File: taiga_halo2/examples/taiga_sudoku/main.rs (hunk 2 above)
Comment (Acentelles) [reply taiga_halo2/examples/taiga_sudoku/main.rs:7]: Which branch do you want to merge first?

Comment (XuyangSong) [reply taiga_halo2/examples/taiga_sudoku/main.rs:7]: Merge this PR to xuyang/sudoku_example branch first
Then I'll refine the `intent_app_vp` and `dealer_intent_vp` in the branch xuyang/sudoku_example

Comment (Acentelles) [reply taiga_halo2/examples/taiga_sudoku/main.rs:7]: Perfect

**Pull request merged**